app/controllers/add-preprint.js

```diff
@@ -1,87 +1,207 @@
-            let preprintMetadata = this.get('store').createRecord('preprint', {
-                    id: path, // TODO: change to guid
-                    attributes: {
-                        title: preprint.title,
-                        abstract: preprint.abstract,
-                        authors: preprint.authors,
-                        date: new Date(),
-                        subject: preprint.subject,
-                        tags: preprint.tags,
-                        journal: preprint.journal,
-                        doi: preprint.doi,
-                        path: path
-                    }
-                });
-
-            preprintMetadata.save().then(() =>
-                this.transitionToRoute('preprints', path)
-            ).catch(error => {
-                console.error(error);
-                this.get('toast').error('Could not save preprint metadata!');
-            });
+        next(name) {
+            // Open next panel
+            this.get('panelActions').open(this.get(`_names.${this.get('_names').indexOf(name) + 1}`));
         },
-        error(ignore, dropzone, file, errorMessage) {
-            console.error(errorMessage);
-            this.get('toast').error(`Error uploading file:${errorMessage}`);
+        changeState(newState) {
+            this.set('state', newState);
         },
-        buildUrl() {
-            return this.get('_url');
+        changeUploadState(newState) {
+            this.set('uploadState', newState);
         },
-        uploadNewPreprintToNewProject(title, abstract, authors, subject, tags, journal, doi) {
-            //Create new public project
-            let node = this.get('store').createRecord('node', {
-                title: title,
+        createProject(resolve) {
+            const node = this.get('store').createRecord('node', {
+                title: this.get('nodeTitle'),
                 category: 'project',
-                description: abstract || null,
                 public: true
             });
-
-            this.set('preprint', {
-                title: title,
-                abstract: abstract,
-                authors: authors,
-                subject: subject,
-                tags: tags,
-                journal: journal,
-                doi: doi
-            });
-
-            // TODO: Add logic for if you're not uploading a file
-            // Upload a new file to the new project
             node.save().then(() => {
-                this.set('_url', `https://test-files.osf.io/file?path=/${this.get('latestFileName')}&nid=${node.id}&provider=osfstorage`);
-                this.set('openModal', false);
-                this.get('resolve')();
-            }).catch(error => {
-                console.error(error);
-                this.get('toast').error('Could not create preprint project!');
+                this.set('preprintNode', node);
+                this._refreshNodes();
+                this.send('startUpload', resolve);
```

Comment:
In terms of timing, it seems possible that the `startUpload` action could begin before `refreshNodes` has ended. This could point to a potential async bug.